assets/js/components/surveys/CurrentSurvey.test.js

```diff
@@ -80,6 +80,147 @@ describe( 'CurrentSurvey', () => {
+			).toBeInTheDocument();
+
+			const STRING_100_CHARACTERS =
+				'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec suscipit auctor dui, id faucibus nisl';
+
+			const STRING_110_CHARACTERS = STRING_100_CHARACTERS + ' rhoncus n';
+
+			// Enter answer text.
+			fireEvent.change( getByPlaceholderText( 'Write here' ), {
+				target: { value: STRING_110_CHARACTERS },
+			} );
+
+			// Input does not have a label so this is the best way to assert the text has been entered correctly
+			expect(
+				getByDisplayValue( STRING_100_CHARACTERS )
+			).toBeInTheDocument();
+		} );
+
+		it( 'should disable submit button when no text is entered', async () => {
+			const { getByPlaceholderText, getByRole } = render(
+				<CurrentSurvey />,
+				{
+					registry,
+				}
+			);
+
+			// Submit button should be disabled if text input is empty.
+			expect( getByRole( 'button', { name: 'Submit' } ) ).toHaveAttribute(
+				'disabled'
+			);
+
+			// Now enter answer text.
+			fireEvent.change( getByPlaceholderText( 'Write here' ), {
+				target: { value: 'Foobar' },
+			} );
+
+			// Submit button should be enabled if text has been entered.
+			expect(
+				getByRole( 'button', { name: 'Submit' } )
+			).not.toHaveAttribute( 'disabled' );
+
+			// Clear and enter input again.
+			fireEvent.change( getByPlaceholderText( 'Write here' ), {
+				target: { value: '' },
+			} );
+			expect( getByRole( 'button', { name: 'Submit' } ) ).toHaveAttribute(
+				'disabled'
+			);
+			fireEvent.change( getByPlaceholderText( 'Write here' ), {
+				target: { value: 'Foobar' },
+			} );
+			expect(
+				getByRole( 'button', { name: 'Submit' } )
+			).not.toHaveAttribute( 'disabled' );
+		} );
+
+		it( 'should submit answer in correct shape', async () => {
+			const { getByPlaceholderText, getByRole, findByText } = render(
+				<CurrentSurvey />,
+				{
+					registry,
+				}
+			);
+
+			expect( fetchMock ).toHaveFetched(
+				/^\/google-site-kit\/v1\/core\/user\/data\/survey-event/
+			);
+
+			expect( fetchMock ).toHaveBeenCalledTimes( 1 );
+
+			fireEvent.change( getByPlaceholderText( 'Write here' ), {
+				target: { value: 'Foobar' },
+			} );
+
+			fireEvent.click( getByRole( 'button', { name: 'Submit' } ) );
+
+			expect( fetchMock ).toHaveBeenCalledTimes( 2 );
+
+			expect( fetchMock ).toHaveFetched(
+				'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
+				{
+					body: {
+						data: {
+							event: {
+								question_answered: {
+									question_ordinal: 1,
+									answer: {
+										answer: 'Foobar',
+									},
+								},
+							},
+							session: {
+								session_id: 'storybook_session',
+								session_token: 'token_12345',
+							},
+						},
+					},
+					credentials: 'include',
+					headers: {
+						Accept: 'application/json, */*;q=0.1',
+						'Content-Type': 'application/json',
+					},
+					method: 'POST',
+				}
+			);
+
+			await findByText( 'Thanks for sharing your thoughts!' );
```

Comment:
Can we associate this with an `expect` statement or something? I'm guessing this is here to ensure this text gets rendered, as the test will time out without it fulfilling the `await`. But that's implicit, so let's explicitly check for that text in the DOM instead.

Reply:
As mentioned somewhere else (on another PR I think) this is copied from the test `should send a 'question_answered' event when a question is answered`

Git blame says you wrote this test :stuck_out_tongue:  (But I'm sure that's just because you merged the Prettier PR :see_no_evil: )


I can change it here and there and every PR *but* I only have this just to not get this error
`Can't perform a React state update on an unmounted component.`